packages/protocol/contracts/L1/ITaikoL1.sol

```diff
@@ -19,6 +19,17 @@ interface ITaikoL1 {
         payable
         returns (TaikoData.BlockMetadata memory meta_, TaikoData.EthDeposit[] memory deposits_);
 
+    /// @notice Proposes a Taiko L2 block (version 2)
+    /// @param _params Block parameters, currently an encoded BlockParams object.
+    /// @param _txList txList data if calldata is used for DA.
+    /// @return meta_ The metadata of the proposed L2 block.
+    function proposeBlockV2(
```

Comment:
Shall we also add `proposeBlockV2` method to `ProverSet`?

Reply:
Done